lib/opam.ml

```diff
@@ -313,23 +313,29 @@ module Url = struct
         else OS.File.must_exist distrib_file >>= fun _ -> assert false
 end
 
-let opam_version () =
-  let v =
+let version v =
+  if Bos_setup.String.is_prefix v ~affix:"1." then (
+    Logs.warn (fun l ->
+        l
+          "opam %s is deprecated, and its support will be dropped in \
+           dune-release 2.0.0, please switch to opam 2.0"
+          v);
+    if String.equal v "1.2.2" then Ok `v1_2_2
+    else R.error_msgf "opam: invalid version %s" v )
```

Comment:
I find the `opam:` prefix a little bit confusing, especially given `opam` isn't responsible for this error but our parsing code is. How about something like `"unknown opam version: %S"` instead?

Reply:
agreed, done.